### ISSUE TITLE
SDD-FIQA results understanding

### ISSUE THREAD

**Issue by lstomy** [Tencent/TFace#20]
Hi,
first of all, thanks for your work.

I decided to test it with pre-trained model and some own real images. I run eval.py demo to check it out.
But I am confused with the results of quality scores.

1. **What ranges of quality score values can be considered as low, medium, high for SDD_FIQA_checkpoints_r50.pth model?**

Referring to the paper, Fig3, I can guess the following value [low: 0 - 35, medium: 36 - 48, high: 49 - 100]. So I wonder to hear your suggestion about it.

2. **Do these ranges depend on the model?** It seems that yes, but correct me if I am wrong.

3. **Did you performed experiments with face occlusion, how its robust on it?**

IMHO, there is no rough thresholds which separate low vs medium vs high sub-ranges, there are some overlaps between them. And these overlaps make me feeling that there is not clear enough quality scores dependency between low, medium and high quality face images.

Thanks~

**Comment by oufuzhao** [comment]
Hi, 
Thanks for your interest in our work. I will answer your questions as follows.

Q1&2: It is hard to divide into three quality levels via thresholds uniformly, including low, medium, and high, for different datasets. On the one hand, there are various potential quality definitions for different testing datasets for the same pre-trained model (e.g., the quality score range of A dataset is [10, 70], but the B dataset one is [30,90]). On the other hand, face IQA algorithms (e.g., SER-FIQ and PFE) focus on learning the order of quality scores rather than the threshold. For ISO/IEC-19794-5 and ICAO-9303 protocols, there is no clear indication of the need for good, medium, and poor quality grades. Therefore, IMHO, if you have to category different classes regarding quality threshold, you can output the quality scores of your dataset and analyze the quality ranges to define thresholds. BTW, the ranges of quality scores can not be divided according to Fig. 3 on our paper: 1). The 100*WD is not equal to the corresponding quality score since the WD needs to be regularized by the max-min operator in Eq. 8. 2). The “Low Quality” and “High Quality” represent the relative quality between the left and right samples rather than the absolute quality definition.

Q3: In the paper, I did not perform this experiment. According to your issue, I randomly pick up some real-world occlusion samples for testing. Moreover, I degree the original image by synthetic distortion and output the quality scores. The results are illustrated in the following figure. If you are interested in this testing, you can further perform the quantitative experiments in terms of occlusion by EVRC.
[image](https://user-images.githubusercontent.com/59477248/124871587-ca111c80-dff6-11eb-88a4-a34845b2206a.png)

Thank :)

**Comment by lstomy** [comment]
Thanks for your kind explanations and quick reply :)

Also, I discovered some other interesting findings such as:

4. As its Face IQA I would expect (or make requirement in problem formulation ;) ) that the approach should output low quality scores (near zero) for random (non-face) images; Currently it outputs any range of quality scores for non-face images. As I understand you didn't make such a restriction on low quality scores values for non-face images, right? IMHO, its reasonable to make such kind of requirement if its Face IQA rather than not generic Visual IQA;

5. When I just take 2-3 more face images of the same person who doesnt move head from consequent frames (other factors like lighting/occlusion, etc. are not changed), the quality score can change gradually, e.g. Q1 = 52, Q2 = 60, Q3 = 48. But sometimes the score values are really stable and reliable. That is only my observation.

6. To be honest, I didnt fully and clearly understand the relation between quality score and face recognition (FR) accuracy. And how FIQA is related/integrated to evaluation protocol of face recognition? What I mean.
Imagine situation. Suppose we have a pair of face images (e.g. different IDs, negative pair), images come from the same dataset. But the first image I1 is high quality, the second image I2 - low/medium quality. Then we run FIQA which outputs quality scores for each image Q1=70 and Q2=32, Q1 > Q2. FR model outputs 2 descriptors: F1 and F2. And for example accordingly to F1, F2 and FR threshold (FRTh) value we get that cos(F1, F2) > FRTh => same (positive), but we know that they are actually different ids => False Positive (FMR in the paper). At same time, we have info that I1 has high quality while I2 low quality score. In this situation its better not to believe to FR but just say that this pair is from different ids (negative), thus making a correct prediction. I wonder to understand the relation between FR accuracy and fact that we have FIQA quality scores. I am not sure if I clearly explained the case, but if you can please comment on it.

Many thanks again!

**Comment by oufuzhao** [comment]
Thanks for your interest in it again. I will try my best to answer your questions in detail in the following:

Q4: This situation is possible since all training samples are aligned face images processed by the face detection tool (i.e., MTCNN). Furthermore, IMHO, face IQA should focus on dealing with the face image rather than face detection. In applications, if you want to achieve your requirement, you can use a face detection approach to make sure the input of the face IQA module. Specifically, if the image does not exist face, output warning, or manually define quality score = 0. Else, align, crop, and resize the face image by the bounding box.

Q5: The factors of face quality contain pose, expression, illumination, occlusion, blur, and so on. The influence of these quality factors is also different for the complex real-world face image. From my observation and analysis, I think the pose and blur usually devote a greater part of the quality score than others in most cases. For illumination, if the light does not greatly affect the gradient information of the face, it will not have a great impact on the recognition system. For occlusion, it may depend on whether the occlusion affects key facial recognition information.

Q6: Uh-huh, I think I can get your information. If you only consider one and two positive/negative pairs, this situation may happen. In a broad sense, if we get Mean(Sum(Negative_Cos[F1, F2(s)])) > FRTh > Mean(Sum(Positive_Cos[F1, F3(s)])), I think the target face image is maybe an outrageous bad case. However, in general, we should get Mean(Sum(Negative_Cos[F1, F2(s)])) < FRTh < Mean(Sum(Positive_Cos[F1, F3(s)])) or FRTh < Mean(Sum(Negative_Cos[F1, F2(s)])) < Mean(Sum(Positive_Cos[F1, F3(s)])). In these two cases, the quality is highly relative to the FR accuracy. That is the reason why the error versus reject curve is widely adopted for performance evaluation in this field.

I hope my answer will help you well.
Best regards :)

**Comment by lstomy** [comment]
Again, many thanks for a detailed response.

ReQ4: you are right that FIQA should focus in its own task. But in real applications FIQA input is face detector output. In real life face detection can be imperfect, dont ideally fit real face bbox (+ false positives can happen) => thus FIQA will have not ideal input.

ReQ5: OK.

ReQ6: Yep, of course all  the situations you mentioned can happen in real situation and they should be treated correspondingly. Could you please share a reference implementation of EVRC you used? Is it public protocol/implementation or you used own-written? I saw that you referenced about EVRC [2, 9, 10, 20, 25] in the paper, but I didnt succeed to find it (implementation) publicly available (may be I just missed smth).

**Comment by Slinene** [comment]
For Q4, the problem is the academic face datasets has few non-face or false positives detection result. In our commercial version, we add such images during training and can be rejected with a low quality score, since their FR sim is very low.
For Q6, EVRC is first defined by us but the evaluation protocol is similar to SER-FIQ

**Comment by Slinene** [comment]
For Q4, the problem is the academic face datasets has few non-face or false positives detection result. In our commercial version, we add such images during training and can be rejected with a low quality score, since their FR sim is very low.
For Q6, EVRC is first defined by us but the evaluation protocol is similar to SER-FIQ

**Comment by lstomy** [comment]
Thanks all authors for having discussion about it. Closing the issue.
